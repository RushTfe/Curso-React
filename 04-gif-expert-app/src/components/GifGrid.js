import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from '../components/GifGridItem'
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

	const { data:images, loading } = useFetchGifs( category );

  	return (
		<>
			<h3 className='animate__animated animate__fadeIn'>{ category }</h3>

			{ loading && <p className='animate__animated animate__flash'>Loading</p> }

			<div className='card-grid'>
				{
					images.map( img => <GifGridItem key={ img.id }{ ...img } />)
				}
			</div>
		</>

  )
}

const getImagesInfo = (images) => images.map(({id, title}) => <li key={id}>{title}</li>);

GifGrid.propTypes = {
	category: PropTypes.string.isRequired
}