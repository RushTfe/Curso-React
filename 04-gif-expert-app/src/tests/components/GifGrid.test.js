import React from 'react';
import '@testing-library/jest-dom'

import { shallow } from 'enzyme';
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas sobre el componentes de GifGrid', () => {

    // let wrapper;
    const category = 'One Punch2';

    // beforeEach(() => {

    //     wrapper = shallow(<GifGrid category={ category } />);

    // });

    test('Debe renderizar el componente correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={ category } />);
        expect( wrapper ).toMatchSnapshot();
        
    });

    test('Debe de mostrar cuando se cargan imÃ¡genes', () => {


        const gifs = [{
            id: 'ABC',
            url: 'https://test/cualquier.jpg',
            title: 'Tituloooo'
        }];
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={ category } />);
        
        //expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists()).toBe( false );

        expect( wrapper.find('GifGridItem').length ).toBe(gifs.length);

    });

})