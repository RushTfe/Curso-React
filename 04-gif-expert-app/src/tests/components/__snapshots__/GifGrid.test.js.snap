// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componentes de GifGrid Debe renderizar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch2
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
